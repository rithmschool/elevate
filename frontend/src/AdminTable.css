.admin-table {
  background: white;
  font-size: 1em;
}

@media only screen and (max-width: 640px)  {
  .admin-table > thead th {
    max-width: 0.5em;
    background: green;
  }
}
