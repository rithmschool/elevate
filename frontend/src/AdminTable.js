import React, { Component } from 'react';
import { Table } from 'react-bootstrap';

class AdminTable extends Component {
  handleClick = (evt) => {
    this.props.getUserDetail(evt);
  }

  render(){
    const keys = Object.keys(this.props.tableObjs[0]);

    return (
      <Table striped bordered hover size="sm" responsive id="users-table">
        <thead>
          <tr>
            { keys.map(key => {
              key = key.replace(/_/g, ' ');
              return <th key={ key } width='55vw'>{ key }</th> 
            })}
          </tr>
        </thead>
        <tbody>
          {this.props.tableObjs.map(item => {
            const itemKeys = Object.keys(item)
            const itemValues = Object.values(item);
            return (
              <tr key={itemValues[0]} onClick={this.handleClick}>
                {itemValues.map((value, index) => {
                  return (
                    <td key={`${itemValues[0]}-${itemKeys[index]}`}>{ value }</td>
                  );
                })}
              </tr>
            );
          })}
        </tbody>
      </Table>
    )
  }
}

export default AdminTable;