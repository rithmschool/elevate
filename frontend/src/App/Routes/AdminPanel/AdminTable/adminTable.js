import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { Table } from "react-bootstrap";

import "./adminTable.css";

const mql = window.matchMedia(`(max-width: 640px)`);
// set the maximum number of table colums for smaller and larger screens
const maxColumCount = mql.matches ? 5 : 12;

class AdminTable extends Component {
  handleClick = evt => {
    const id = evt.target.parentElement.id;
    this.props.history.push(`/admin/users/${id}`);
  };

  createTableHeader() {
    const keys = Object.keys(this.props.tableObjs[0]);

    return (
      <tr>
        {keys
          .map(key => {
            // Remove underscore from key name
            key = key.replace(/_/g, " ");

            return <th key={key}>{key}</th>;
          })
          .filter((key, idx) => idx < maxColumCount)}
      </tr>
    );
  }

  createTableBody() {
    const table = this.props.tableObjs.map(item => {
      const itemKeys = Object.keys(item);
      const itemValues = Object.values(item);
      return this.createTableRows(itemKeys, itemValues);
    });

    return table;
  }

  concantinateText(value) {
    // concatinate at 6 characters for small screen and 25 for large
    if (typeof value === "string") {
      if (mql.matches) {
        // small screen
        if (value.slice(0, 2) === "19" || value.slice(0, 2) === "20") {
          // only show year on small strings
          value = value.slice(0, 4);
        }
        if (value.length > 9) {
          value = value.slice(0, 6) + "...";
        }
      } else {
        // larger screen
        if (value.length > 30) {
          value = value.slice(0, 25) + "...";
        }
      }
    }

    return value;
  }

  createTableRows(keys, values) {
    return (
      <tr key={values[0]} onClick={this.handleClick} id={values[0]}>
        {values
          .map((value, index) => {
            value = this.concantinateText(value);

            return <td key={`${values[0]}-${keys[index]}`}>{value}</td>;
          })
          .filter((value, idx) => idx < maxColumCount)}
      </tr>
    );
  }

  render() {
    const tableType = this.props.tableObjs[0].hasOwnProperty("company")
      ? "users"
      : "questions";
    return (
      <div className="admin-table">
        <Table
          striped
          bordered
          hover
          size="sm"
          responsive
          id={tableType + "-table"}
        >
          <thead>{this.createTableHeader()}</thead>
          <tbody>{this.createTableBody()}</tbody>
        </Table>
      </div>
    );
  }
}

export default withRouter(AdminTable);
