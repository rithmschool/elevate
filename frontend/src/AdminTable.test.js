import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import AdminTable from "./adminTable";

describe("AdminTable", function() {
  let wrapper;
  let users = [
    {
      id: 17,
      email: "testadmin@test.com",
      is_admin: true,
      first_name: "admin",
      last_name: "test",
      current_company: "testcompany",
      hire_date: "2018-06-23",
      needs: "To test user data",
      goals: "Test pass"
    }
  ];

  beforeEach(() => {
    wrapper = mount(<AdminTable tableObjs={users} />);
  });

  it("renders without crashing", function() {
    // Mounted in beforeEach above
  });

  it("matches snapshot", function() {
    const serialized = toJson(wrapper);

    expect(serialized).toMatchSnapshot();
  });

  it("has props", function() {
    expect(wrapper.props("tableObjs")).toEqual({
      tableObjs: [
        {
          current_company: "testcompany",
          email: "testadmin@test.com",
          first_name: "admin",
          goals: "Test pass",
          hire_date: "2018-06-23",
          id: 17,
          is_admin: true,
          last_name: "test",
          needs: "To test user data"
        }
      ]
    });
  });

  it("creates a table", function() {
    expect(wrapper.find("table")).toHaveLength(1);
  });
});
