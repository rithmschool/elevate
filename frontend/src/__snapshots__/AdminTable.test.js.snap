// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminTable matches snapshot 1`] = `
<AdminTable
  tableObjs={
    Array [
      Object {
        "current_company": "testcompany",
        "email": "testadmin@test.com",
        "first_name": "admin",
        "goals": "Test pass",
        "hire_date": "2018-06-23",
        "id": 17,
        "is_admin": true,
        "last_name": "test",
        "needs": "To test user data",
      },
    ]
  }
>
  <div
    className="admin-table"
  >
    <Bootstrap(Table)
      bordered={true}
      hover={true}
      id="undefined-table"
      responsive={true}
      size="sm"
      striped={true}
    >
      <Table
        bordered={true}
        bsPrefix="table"
        hover={true}
        id="undefined-table"
        responsive={true}
        size="sm"
        striped={true}
      >
        <div
          className="table-responsive"
        >
          <table
            className="table table-sm table-striped table-bordered table-hover"
            id="undefined-table"
          >
            <thead>
              <tr>
                <th
                  key="id"
                >
                  id
                </th>
                <th
                  key="email"
                >
                  email
                </th>
                <th
                  key="is admin"
                >
                  is admin
                </th>
                <th
                  key="first name"
                >
                  first name
                </th>
                <th
                  key="last name"
                >
                  last name
                </th>
                <th
                  key="current company"
                >
                  current company
                </th>
                <th
                  key="hire date"
                >
                  hire date
                </th>
                <th
                  key="needs"
                >
                  needs
                </th>
                <th
                  key="goals"
                >
                  goals
                </th>
              </tr>
            </thead>
            <tbody
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <tr
                id={17}
                key="17"
                onClick={[Function]}
              >
                <td
                  key="17-id"
                >
                  17
                </td>
                <td
                  key="17-email"
                >
                  testadmin@test.com
                </td>
                <td
                  key="17-is_admin"
                >
                  <Component />
                </td>
                <td
                  key="17-first_name"
                >
                  admin
                </td>
                <td
                  key="17-last_name"
                >
                  test
                </td>
                <td
                  key="17-current_company"
                >
                  testcompany
                </td>
                <td
                  key="17-hire_date"
                >
                  2018-06-23
                </td>
                <td
                  key="17-needs"
                >
                  To test user data
                </td>
                <td
                  key="17-goals"
                >
                  Test pass
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </Table>
    </Bootstrap(Table)>
  </div>
</AdminTable>
`;
